### PR TITLE
Introduction of new Gemini API to sync graph data

### PR DESCRIPTION
**Pull request by yzainee**

### CODE REVIEW

File: src/rest_api.py [hunk 1]
```diff
@@ -44,6 +53,55 @@ def liveness():
...
+    """
+    resp = {
+        "success": True,
+        "message": "Sync operation started ->"
+    }
+
+    input_json = request.get_json()
+    logger.info("sync-graph-data called with the input {i}".format(i=input_json))
+    non_cve_sync = input_json.get('non_cve_sync', False)
+    if non_cve_sync == "true":
```
Comment (samuzzal-choudhury) [inline src/rest_api.py]: @yzainee  as per the swagger spec, `'non_cve_sync'` is a boolean field and should not be compared with a string `"true"`.

The boolean value comparison in Python happens like `if non_cve_sync:` or like `if not non_cve_sync`.

File: src/rest_api.py [hunk 2]
```diff
@@ -44,6 +53,55 @@ def liveness():
...
+            logger.error("Incorrect data.. Send cve_ecosystem for non_cve_sync operation")
+            return flask.jsonify(resp), 400
+        url = "http://{host}:{port}/{endpoint}".format(host=_SERVICE_HOST,
+                                                       port=_SERVICE_PORT,
+                                                       endpoint=_CVE_SYNC_ENDPOINT)
+        logger.info("Calling non cve sync with {i}".format(i=cve_ecosystem))
+        _session.post(url, json=cve_ecosystem)
+        resp['message'] = resp['message'] + " for non cve version"
+    latest_version_sync = input_json.get('latest_version_sync', False)
+    if latest_version_sync == "true":
```
Comment (samuzzal-choudhury) [inline src/rest_api.py]: @yzainee as per the swagger spec, `'latest_version_sync'` is a boolean field which should not be compared with a string `"true"`.

The boolean value comparison in Python happens like `if latest_version_sync` or like `if not latest_version_sync `.

File: tests/test_rest_api.py [hunk 3]
```diff
@@ -154,6 +154,40 @@ def test_liveness_endpoint_wrong_http_method(client):
     assert response.status_code == 405
 
 
+@patch("requests.Session.post", return_value="")
+def test_sync_data1(_mock1, client):
+    """Test the /api/v1/sync-graph-data endpoint."""
+    url = api_route_for("sync-graph-data")
+    inp = {
+        "non_cve_sync": "false",
+        "latest_version_sync": "true",
```
Comment (samuzzal-choudhury) [inline tests/test_rest_api.py]: @yzainee  inputs for `non_cve_sync` and `latest_version_sync` should be boolean and not string. The same comment applies for the below scenarios as well.

File: src/rest_api.py [hunk 4]
```diff
@@ -44,6 +53,55 @@ def liveness():
...
+def sync_data():
+    """
+    Endpoint for carrying a sync of graph db data.
+
+    Takes in value to either call sync of non cve version
+    or latest version or both.
+
+    valid input: {
+        "non_cve_sync": True / False,
+        "latest_version_sync": True / false,
```
Comment (samuzzal-choudhury) [inline src/rest_api.py]: @yzainee  typo. should have been `true / false`.

File: tests/test_rest_api.py [hunk 5]
```diff
@@ -154,6 +154,40 @@ def test_liveness_endpoint_wrong_http_method(client):
     assert response.status_code == 405
 
 
+@patch("requests.Session.post", return_value="")
+def test_sync_data1(_mock1, client):
+    """Test the /api/v1/sync-graph-data endpoint."""
+    url = api_route_for("sync-graph-data")
+    inp = '''{
+        "non_cve_sync": true,
```
Comment (samuzzal-choudhury) [inline tests/test_rest_api.py]: @yzainee the value for `non_cve_sync` and `latest_version_sync` has to be boolean throughout.

Comment (yzainee) [reply tests/test_rest_api.py]: Maan gaye... aapki paarkhi nazar :)

File: src/rest_api.py (hunk 4 above)
Comment (yzainee) [reply src/rest_api.py]: changed

**Pull request merged**